erpnext_affirm/erpnext_affirm_integration/doctype/affirm_settings/affirm_settings.py

```diff
@@ -163,12 +163,17 @@ def build_checkout_data(**kwargs):
 
 	items = []
 	discounts = {}
+	billing_address = None
 	shipping_address = None
 
-	if ref_doc.shipping_address:
-		shipping_address = frappe.get_doc("Address", ref_doc.shipping_address)
-	else:
-		shipping_address = frappe.get_doc("Address", ref_doc.billing_address)
+	if order_doc.get("customer_address"):
+		billing_address = frappe.get_doc("Address", order_doc.customer_address)
+
+	if order_doc.get("shipping_address_name"):
+		shipping_address = frappe.get_doc("Address", order_doc.shipping_address_name)
+
+	if not shipping_address:
+		shipping_address = frappe.get_doc("Address", frappe.get_value("Affirm Settings", "Affirm Settings", "pickup_address"))
```

Comment:
This will break if no pickup address is set, or (if we're looking at general adoption) if the company doesn't have a pickup address.

Some thoughts:
- Would it be better to default the pickup address to the Company Address?
- For general use, I think it would be better if the pickup address(es) are set and available through the shopping cart, rather than being linked to Affirm specifically.

Reply:
No. Company Address might not be the pickup address in all cases. Agreed with your second point, in case they can offer multiple pickup points. but that's a feature in itself. This looks good for now